pytorch_lightning/lite/wrappers.py

```diff
@@ -65,23 +65,27 @@ def step(self, closure: Optional[Callable] = None) -> Any:
 
 
 class _LiteModule(DeviceDtypeModuleMixin):
-    def __init__(self, module: nn.Module, precision_plugin: PrecisionPlugin) -> None:
+    def __init__(self, forward_module: nn.Module, precision_plugin: PrecisionPlugin, original_module: Optional[nn.Module] = None) -> None:
         """The LiteModule is a thin wrapper around the :class:`torch.nn.Module` and handles precision / autocast
         automatically for the forward pass.
 
         The underlying wrapped module can be accessed via the property :attr:`module`.
 
         Args:
-            module: The module to wrap
+            forward_module: The module to wrap the ``forward`` method on.
             precision_plugin: Reference to the precision plugin for handling precision context
+            original_module: The original, unmodified module as passed into the
+                :meth:`pytorch_lightning.lite.lite.LightningLite.setup` method. This is needed when accessing
+                attributes through the :attr:`module` property.
         """
         super().__init__()
-        self._module = module
+        self._forward_module = forward_module
+        self._original_module = original_module
         self._precision_plugin = precision_plugin
 
     @property
     def module(self) -> nn.Module:
-        return self._module
+        return self._original_module or self._forward_module
```

Comment:
Should we really have a separate reference here? Can't we just call the unwrap function, we used to have?

Reply:
The "unwrap" function is implemented for LightningModule and has a stopping condition with an isinstance check against LightningModule. This can't be so easily converted to an equivalent unwrapping method for pure nn.Modules.
Afaik, this is the safest way to guarantee we actually obtain the original untouched model that was passed in to the Lite.setup method.

Reply:
Off topic to the PR: I find the unwrapping logic we have in Lightning hard to follow. I would like that we clean it up someday

This solution by Adrian is clean and explicit IMO

Reply:
I agree, The only motivation for asking was consistency so that when we clean it in one place, it would still all behave the same :)